contracts/EscrowDistributor.sol

```diff
@@ -0,0 +1,63 @@
+// SPDX-License-Identifier: MIT
+pragma solidity ^0.8.0;
+
+import "@openzeppelin/contracts/token/ERC20/IERC20.sol";
+
+import "./interfaces/IRewardEscrow.sol";
+
+contract EscrowDistributor {
+    /// @notice kwenta token contract
+    address public immutable kwentaAddr;
```

Comment:
if you save this as IERC20 vs address, would that save gas? Would be good to know. That way you don't have to do the `IERC20(kwenta).transferFrom()` and can just do it `kwenta.transferFrom()`.

May not make a difference

Reply:
Yea I did have it like that and then thought it might be handy to store address if ever anyone wanted to read which token address it was, but probably not really needed. Is there a simple way you compare gas, can you get that gas value when calling in a test?

Reply:
I believe even if you do store an address as a contract type, it will still return the address offchain (ie. on etherscan)

Example of StakingRewards if you try to read rewardEscrow or supplySchedule: https://goerli-optimism.etherscan.io/address/0x1653a3a3c4ccee0538685f1600a30df5e3ee830a#readContract

Reply:
ok good to know thanks, now storing both rewardEscrow and kwenta as contract instances instead of addresses

Reply:
So preference I suppose. And as far as testing gas costs, hardhat has a plugin. Foundry reports it by default which is what i was using to test out my theory haha

Reply:
Good to know